### PR TITLE
Refactor authInfoTest.ts

### PR DESCRIPTION
**Pull request by mdonnalley**
- Update `authInfoTest.ts` to rely on `testSetup`. Also modestly increases test coverage of `AuthInfo`. The original tests were all moved to `oldAuthInfoTest.ts`, which we can delete before merging
Before
```
----------------------------------|---------|----------|---------|---------|------------------------
File                              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------------------------------|---------|----------|---------|---------|------------------------
authInfo.ts                     |   92.64 |     82.4 |   97.92 |   93.42 | ...0,986-987,1127-1129
```

After
```
----------------------------------|---------|----------|---------|---------|------------------------
File                              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------------------------------|---------|----------|---------|---------|------------------------
authInfo.ts                     |    94.8 |    85.41 |   97.96 |   95.63 | ...469,660,798,883,894
```

- Update tests to use `shouldThrow` and `shouldThrowSync` wherever appropriate
- Remove `getString` and `getNumber` from codebase

@W-11274518@
@W-11288647@

### CODE REVIEW

**Review by peternhale — APPROVED** [review]
I love all the red in the diffs. Great job!

File: src/org/authInfo.ts [hunk 1]
```diff
@@ -709,7 +712,7 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
     this.stateAggregator = await StateAggregator.getInstance();
 
     const username = this.options.username;
-    const authOptions = this.options.oauth2Options || this.options.accessTokenOptions;
+    const authOptions = (this.options.oauth2Options || this.options.accessTokenOptions) as AuthOptions;
```
Comment (mshanemc) [inline src/org/authInfo.ts:715]:
```suggestion
    const authOptions = (this.options.oauth2Options ?? this.options.accessTokenOptions) as AuthOptions;
```

File: src/org/authInfo.ts [hunk 2]
```diff
@@ -724,7 +727,7 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
       }
     }
 
-    const oauthUsername = username || getString(authOptions, 'username');
+    const oauthUsername = username || authOptions?.username;
```
Comment (mshanemc) [inline src/org/authInfo.ts:730]:
```suggestion
    const oauthUsername = username ?? authOptions?.username;
```

File: src/org/authInfo.ts [hunk 3]
```diff
@@ -756,10 +759,9 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
     }
   }
 
-  private getInstanceUrl(options: unknown, aggregator: ConfigAggregator) {
+  private getInstanceUrl(options: AuthOptions, aggregator: ConfigAggregator) {
     const instanceUrl =
-      getString(options, 'instanceUrl') ||
-      (aggregator.getPropertyValue(OrgConfigProperties.ORG_INSTANCE_URL) as string);
+      options?.instanceUrl || (aggregator.getPropertyValue(OrgConfigProperties.ORG_INSTANCE_URL) as string);
```
Comment (mshanemc) [inline src/org/authInfo.ts:764]:
```suggestion
      options?.instanceUrl ?? (aggregator.getPropertyValue(OrgConfigProperties.ORG_INSTANCE_URL) as string);
```

Comment (mshanemc) [inline src/org/authInfo.ts:764]: side comment, not for this PR

wouldn't it be nice if getPropertyValue knew the types of at least the known properties?

File: src/org/authInfo.ts [hunk 4]
```diff
@@ -756,10 +759,9 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
...
     return instanceUrl || SfdcUrl.PRODUCTION;
```
Comment (mshanemc) [inline src/org/authInfo.ts:765]:
```suggestion
    return instanceUrl ?? SfdcUrl.PRODUCTION;
```

File: src/org/authInfo.ts [hunk 5]
```diff
@@ -884,15 +886,19 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
...
   private async authJwt(options: OAuth2Config): Promise<AuthFields> {
     if (!options.clientId) {
       throw messages.createError('missingClientId');
     }
-    const privateKeyContents = await fs.promises.readFile(ensure(options.privateKey), 'utf8');
+    const privateKeyContents = await this.readJwtKey(ensureString(options.privateKey));
     const { loginUrl = SfdcUrl.PRODUCTION } = options;
     const url = new SfdcUrl(loginUrl);
-    const createdOrgInstance = getString(options, 'createdOrgInstance', '').trim().toLowerCase();
+    const createdOrgInstance = (this.getFields().createdOrgInstance || '').trim().toLowerCase();
```
Comment (mshanemc) [inline src/org/authInfo.ts:901]:
```suggestion
    const createdOrgInstance = (this.getFields().createdOrgInstance ?? '').trim().toLowerCase();
```

File: src/org/authInfo.ts [hunk 6]
```diff
@@ -1090,17 +1095,15 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
    * @param response
    * @private
    */
-  private throwUserGetException(response: unknown) {
+  private throwUserGetException(response: { body?: string }) {
     let errorMsg = '';
-    const bodyAsString = getString(response, 'body', JSON.stringify({ message: 'UNKNOWN', errorCode: 'UNKNOWN' }));
+    const bodyAsString = response.body || JSON.stringify({ message: 'UNKNOWN', errorCode: 'UNKNOWN' });
```
Comment (mshanemc) [inline src/org/authInfo.ts:1100]:
```suggestion
    const bodyAsString = response.body ?? JSON.stringify({ message: 'UNKNOWN', errorCode: 'UNKNOWN' });
```

File: src/org/authInfo.ts [hunk 7]
```diff
@@ -1090,17 +1095,15 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
    * @param response
    * @private
    */
-  private throwUserGetException(response: unknown) {
+  private throwUserGetException(response: { body?: string }) {
```
Comment (mshanemc) [inline src/org/authInfo.ts:1098]: 😄

File: src/org/connection.ts [hunk 8]
```diff
@@ -473,7 +464,7 @@ export class Connection<S extends Schema = Schema> extends JSForceConnection<S>
   private async loadInstanceApiVersion(): Promise<Nullable<string>> {
     const authFileFields = this.options.authInfo.getFields();
     const lastCheckedDateString = authFileFields.instanceApiVersionLastRetrieved;
-    let version = getString(authFileFields, 'instanceApiVersion');
+    let version = authFileFields.instanceApiVersion;
```
Comment (mshanemc) [inline src/org/connection.ts:467]: 😄

File: src/stateAggregator/accessors/orgAccessor.ts [hunk 9]
```diff
@@ -90,6 +102,12 @@ export abstract class BaseOrgAccessor<T extends ConfigFile, P extends ConfigCont
     return this.getAll(decrypt);
   }
 
+  /**
```
Comment (mshanemc) [inline src/stateAggregator/accessors/orgAccessor.ts:105]: oh, these are nice, too

**Review by mshanemc — APPROVED** [review]
I'd approve it several extra times if I could.  This is great!

**Pull request merged**